src/frontend/src/routes/(new-styling)/login/+page.ts

```diff
@@ -1,5 +1,19 @@
 import type { PageLoad } from "./$types";
 
 export const load: PageLoad = ({ url }) => {
-  return { next: url.searchParams.get("next") };
+  // Sanitize the next search param, only accept relative paths
```

Comment:
(repeating an idea fro above) 

The best sanitization is also a (unique) constructor of an object of type `ValidT` for your daat of type `T`.